src/main/java/seedu/address/model/moduleclass/UniqueModuleClassList.java

```diff
@@ -0,0 +1,156 @@
+package seedu.address.model.moduleclass;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.moduleclass.exceptions.DuplicateModuleClassException;
+import seedu.address.model.moduleclass.exceptions.ModuleClassNotFoundException;
+
+/**
+ * A list of {@code ModuleClass} that enforces uniqueness between its elements and does not allow nulls.
+ * A {@code ModuleClass} is considered unique by comparing using {@code ModuleClass#isSameModuleClass(ModuleClass)}.
+ * As such, adding and updating of {@code ModuleClass} uses ModuleClass#isSameModuleClass(ModuleClass) for equality
+ * so as to ensure that the {@code ModuleClass} being added or updated is unique in terms of identity in the
+ * UniqueModuleClassList. However, the removal of a {@code ModuleClass} uses ModuleClass#equals(Object) so as to ensure
+ * that the {@code ModuleClass} with exactly the same fields will be removed.
```

Comment:
See if its possible to trim down this part. I took a couple of tries before I understood what was going on

Reply:
This JavaDoc was directly adapted from a corresponding JavaDoc in the existing UniqueStudentList class, so I would be more inclined to retain the current expression since it is consistent with existing code.

Reply:
Ok lets retain it